Fuzzlyn/Methods/FuncBodyGenerator.cs

```diff
@@ -342,37 +358,106 @@ private StatementSyntax GenReturn()
 
     private StatementSyntax GenLoop()
     {
-        string varName = $"var{_varCounter++}";
-        ScopeValue indVar = new(_types.GetPrimitiveType(SyntaxKind.IntKeyword), IdentifierName(varName), -(_scope.Count - 1), true);
+        if (_random.FlipCoin(_random.Options.ForLoopProb))
+        {
+            return GenForLoop();
+        }
+        else
+        {
+            return GenDoLoop();
+        }
+    }
+
+    private StatementSyntax GenForLoop()
+    {
+        string varName = $"lvar{_varCounter++}";
+        SyntaxKind op = (SyntaxKind)Options.LoopIndexTypeDist.Sample(_random.Rng);
+        PrimitiveType indexPrimType = _types.GetPrimitiveType(op);
+        ScopeValue indexVar = new(indexPrimType, IdentifierName(varName), -(_scope.Count - 1), true);
+        (ExpressionSyntax lowerBound, ExpressionSyntax upperBound) = LiteralGenerator.GenPrimitiveLiteralLoopBounds(_random, indexPrimType);
```

Comment:
I try hard to avoid Fuzzlyn generating programs that diverge or timeout. Isn't this going to be very likely to generate such programs?

Reply:
No: it's always generating a loop of two iterations (currently).